tests/unit/utils_test.py

```diff
@@ -42,6 +42,23 @@ def test_full_query():
         assert key in pipeline[i].keys() and len(pipeline[i].keys()) == 1
 
 
+def test_filter_on_categoricals():
+    import itertools
+
+    query = MongoDBQuery()
+    query.add_standard_match(
+        date_min="1999-12-13",
+        date_max="2021-12-10",
+        min_effectif=10,
+        sirets=["44937840500012"],
+        categorical_filters={"region": ["Bourgogne-Franche-Comté"]},
+    )
+    pipeline = query.to_pipeline()
+    assert "$match" in pipeline[0].keys()
+    test_filters = [filter.keys() for filter in pipeline[0]["$match"]["$and"]]
+    assert "value.region" in list(itertools.chain(*test_filters))
```

Comment:
You can do it in 1 line and without itertools

```suggestion
    test_filters = [key for dicts in pipeline[0]["$match"]["$and"] for key, val in dicts.items()]
    assert "value.region" in test_filters
```

See [this SO answer](https://stackoverflow.com/a/11399555)

Reply:
Bien vu! Je me disais bien que ça devait possible sans itertools, j'aurais dû fouiller un peu plus ^^'